### PR TITLE
ESC-896 Resolve error message issues on the 'Add a reference' page while adhering to design system guidelines.

### PR DESCRIPTION
**Pull request by pbyrne84**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ package object types extends SimpleJson {
   type TraderRef = String @@ TraderRef.Tag
   object TraderRef
       extends RegexValidatedString(
-        regex = """.{0,35}"""
+        regex = """.{1,400}"""
```
Comment (pbyrne84) [inline app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala:63]: This now matches the frontend validation else an empty string will blow the front end up.

Really there needs to be a single source of truth.

File: test/uk/gov/hmrc/eusubsidycompliance/models/SubsidyUpdateSpec.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmrc.eusubsidycompliance.models
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.Json
+import uk.gov.hmrc.eusubsidycompliance.models.types.{EORI, EisSubsidyAmendmentType, SubsidyAmount, SubsidyRef, TraderRef, UndertakingRef}
+
+import java.time.LocalDate
+
+class SubsidyUpdateSpec extends AnyWordSpec with Matchers {
```
Comment (pbyrne84) [inline test/uk/gov/hmrc/eusubsidycompliance/models/SubsidyUpdateSpec.scala:10]: Handy having a spec to see the json contracts we allow. Relying on case classes to generate test data means are contracts are very hard to work out, especially nested case classes.

File: app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala (hunk 1 above)
Comment (masahmad) [reply app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala:63]: Patrick,  in my latest PR  the regex is actually updated to the folowing...

 regex = """[A-Za-z0-9 ]{1,400}"""

https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/357/files#diff-dd0d4530e38e33f61c2e5d4d3a367c1cbbe356c51c478e4e2d68eefaa73d0344

**Comment by habeeb01** [comment]
please address the PR bot comments

File: app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala (hunk 1 above)
Comment (habeeb01) [reply app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala]: @masahmad, I think `\s` is what is used for representing whitespace. Also Mas' regex allows whitespace/numbers/letters but no special characters. @pbyrne84, your regex allows any characters. We should confirm (from ETMP) if special characters are allowed and out frontend and our backend validation should match.

File: app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala [hunk 3]
```diff
@@ -60,7 +60,7 @@ package object types extends SimpleJson {
   type TraderRef = String @@ TraderRef.Tag
   object TraderRef
       extends RegexValidatedString(
-        regex = """.{0,35}"""
+        regex = """[A-Za-z0-9 ]{1,400}"""
```
Comment (habeeb01) [inline app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala:63]: `\s` is the regex for white space, Ive asked Mas to update on the frontend. Can we do the same on the backend.

Comment (habeeb01) [reply app/uk/gov/hmrc/eusubsidycompliance/models/types/package.scala:63]: please ignore the above comment. `\s` matches tabs as well as other whitespace chars.

**Review by habeeb01 — APPROVED** [review]

**Pull request merged**